com.minres.coredsl/src/com/minres/coredsl/scoping/CoreDslScopeProvider.xtend

```diff
@@ -79,6 +80,14 @@ class CoreDslScopeProvider extends AbstractDeclarativeScopeProvider { //Abstract
         else
             IScope.NULLSCOPE
     }
+    
+    def IScope scope_Variable(IterationStatement context, EReference reference) {
+        val surrScope = (context as BlockItem).scope_Variable(reference)
```

Comment:
This should be merged into the preceeding  scope_Variable(BlockItem context, EReference reference) since IterationStatement inherits of BlockItem. So just treat it as special case of BlockItem.

Other option is to specialize scope_Variable(BlockItem context, EReference reference) for the various sub-classes of BlockItem (Declaration, CompoundStatement, ...)

Reply:
Done.